### PR TITLE
BufferBase: Split validation in helper methods

### PR DESCRIPTION
**Pull request by Kangz**
PTAL

### CODE REVIEW

File: src/backend/Buffer.cpp [hunk 1]
```diff
@@ -148,26 +141,43 @@ namespace backend {
         mMapUserdata = 0;
     }
 
-    bool BufferBase::ValidateMapBase(uint32_t start,
-                                     uint32_t size,
-                                     dawn::BufferUsageBit requiredUsage) {
+    MaybeError BufferBase::ValidateSetSubData(uint32_t start, uint32_t count) const {
+        if (start + count > GetSize()) {
+            DAWN_RETURN_ERROR("Buffer needs the transfer dst usage bit");
```
Comment (SenorBlanco) [inline src/backend/Buffer.cpp]: This error message looks wrong.

Comment (Kangz) [reply src/backend/Buffer.cpp]: Done

File: src/backend/Buffer.cpp [hunk 2]
```diff
@@ -148,26 +141,43 @@ namespace backend {
         mMapUserdata = 0;
     }
 
-    bool BufferBase::ValidateMapBase(uint32_t start,
-                                     uint32_t size,
-                                     dawn::BufferUsageBit requiredUsage) {
+    MaybeError BufferBase::ValidateSetSubData(uint32_t start, uint32_t count) const {
+        if (start + count > GetSize()) {
```
Comment (kainino0x) [inline src/backend/Buffer.cpp]: Handle int overflow? (or add a TODO)

File: src/backend/Buffer.cpp [hunk 3]
```diff
@@ -148,26 +141,43 @@ namespace backend {
...
+        }
+
+        return {};
+    }
+
+    MaybeError BufferBase::ValidateMap(uint32_t start,
+                                       uint32_t size,
+                                       dawn::BufferUsageBit requiredUsage) const {
         // TODO(cwallez@chromium.org): check for overflows.
         if (start + size > GetSize()) {
```
Comment (kainino0x) [inline src/backend/Buffer.cpp]: Here too.

File: src/backend/Buffer.cpp (hunk 2 above)
Comment (Kangz) [reply src/backend/Buffer.cpp]: Added a TODO

File: src/backend/Buffer.cpp (hunk 3 above)
Comment (Kangz) [reply src/backend/Buffer.cpp]: There was a TODO already.

**Pull request merged**